### PR TITLE
Fix StreamPool_MultipleStreamsInSequence_PooledStreamReused

### PR DESCRIPTION
**Pull request by JamesNK**
https://github.com/dotnet/aspnetcore/issues/34685

### CODE REVIEW

File: src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs [hunk 1]
```diff
@@ -127,8 +127,9 @@ public void Initialize(Http3StreamContext context)
             _frameWriter.Reset(context.Transport.Output, context.ConnectionId);
         }
 
-        public void InitializeWithExistingContext(IDuplexPipe transport)
+        public void InitializeWithExistingContext(IDuplexPipe transport, string connectionId)
```
Comment (halter73) [inline src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs:130]: Should this just take a Http3StreamContext?

Comment (JamesNK) [reply src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs:130]: The Http3StreamContext would need to be made public on Http3Stream to be able to pass it back in again.

The current code follows what Http2Stream does. The difference is there are a couple of properties that need to be updated on the context.

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs [hunk 2]
```diff
@@ -219,45 +220,69 @@ public async Task StreamPool_MultipleStreamsInSequence_PooledStreamReused()
...
-            await requestStream.SendHeadersAsync(headers);
-            await requestStream.SendDataAsync(Encoding.ASCII.GetBytes("Hello world 1"), endStream: true);
+            Assert.Same(streamContext1, streamContext2);
+        }
 
-            Assert.False(requestStream.Disposed);
+        [Theory]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(1000)]
```
Comment (halter73) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs:232]: Are these numbers significant for any reason? Why does testing two requests as above not give us sufficient coverage? I know these are really fast, but how long does it take to complete 1000 requests in sequence?

**Review by halter73 — APPROVED** [review]

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs (hunk 2 above)
Comment (JamesNK) [reply src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3ConnectionTests.cs:232]: No significance. 1000 takes less than a second. I wanted a test that stresses pooling and makes thread-safety issues fail reliable.

**Pull request merged**